File: kotlin/reverse-string/src/main/kotlin/ReverseString.kt
//fun reverse( inputString: String): String {
////    return inputString.toCharArray().reversedArray().joinToString("")
//    return inputString.toCharArray().reversed().joinToString("")
//}

//fun reverse( inputString: String ) = inputString.reversed()


//fun reverse(str:String): String { return str.split(" ").reduce{acc, x -> x + " " + acc}}

//fun reverse(str:String): String { return str.fold(""){ acc, x -> x + acc } }

fun reverse(str: String) = str.reduce{ acc, x -> x + acc}
fun reverse(str: String) = str.fold(""){ acc, x -> x + acc}
fun reverse(str: String) = str.reversed()

File: dart/reverse-string/lib/reverse_string.dart
String reverse([String inputString = '']) => inputString.split('').reversed.join('');


String reverse(String str) {
  return str.length > 0
      ? str.split('').reduce((accum, letter) => letter + accum)
      : str;
}


String reverse([String str = '']) {
  // Put your code here
    List<String> reversed = [];

  for (String char in str.split('')) {
    reversed.insert(0, char);
  }

  return reversed.join('');
}


String reverse(String input) {
  return input.runes.fold("", (String a, int b) => String.fromCharCode(b) + a);
}


